scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -69,51 +97,119 @@ private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl s
-        DocumentConfig documentConfig = DocumentConfig.from(documentId, newImports.toString(), documentName);
+        DocumentId documentId = DocumentId.create(generatedDocumentName, defaultModuleId);
+        DocumentConfig documentConfig = DocumentConfig.from(documentId, newImports.toString(), generatedDocumentName);
         defaultModule.modify().addDocument(documentConfig).apply();
 
         project.currentPackage().ballerinaToml().ifPresent(ballerinaToml -> {
             String tomlFileContent = ballerinaToml.tomlDocument().textDocument().toString();
             ballerinaToml.modify().withContent(tomlFileContent + tomlDependencies).apply();
         });
 
-        List<ScannerContext> scannerContextList = new ArrayList<>();
+        PackageResolution packageResolution = project.currentPackage().getResolution();
+        ResolvedPackageDependency rootPkgNode = new ResolvedPackageDependency(project.currentPackage(),
+                PackageDependencyScope.DEFAULT);
+        List<Package> directDependencies = packageResolution.dependencyGraph().getDirectDependencies(rootPkgNode)
+                .stream().map(ResolvedPackageDependency::packageInstance).toList();
+        Map<String, List<Rule>> externalAnalyzers = new HashMap<>();
+        for (Package pkgDependency : directDependencies) {
+            PackageManifest pkgManifest = pkgDependency.manifest();
+            PackageOrg org = pkgManifest.org();
+            PackageName name = pkgManifest.name();
+            if (pkgManifest.compilerPluginDescriptor().isEmpty() || !analyzerDescriptors
+                    .contains(org.value() + FORWARD_SLASH + name.value())) {
+                continue;
+            }
 
-        // TODO: Implement the logic to pass scanner contexts to compiler plugins and perform external static analysis.
+            CompilerPluginDescriptor pluginDesc = pkgManifest.compilerPluginDescriptor().get();
+            List<String> jarPaths = new ArrayList<>();
+            pluginDesc.dependencies().forEach(dependency -> {
+                jarPaths.add(dependency.getPath());
+            });
 
-        List<Issue> externalIssues = new ArrayList<>(scannerContextList.size());
-        for (ScannerContext scannerContext : scannerContextList) {
-            ReporterImpl reporter = (ReporterImpl) scannerContext.getReporter();
-            externalIssues.addAll(reporter.getIssues());
+            List<URL> jarUrls = new ArrayList<>();
+            jarPaths.forEach(jarPath -> {
+                try {
+                    URL jarUrl = Path.of(jarPath).toUri().toURL();
+                    jarUrls.add(jarUrl);
+                } catch (MalformedURLException ex) {
+                    throw new RuntimeException(ex);
+                }
+            });
+
+            URLClassLoader ucl = AccessController.doPrivileged((PrivilegedAction<URLClassLoader>) () ->
+                    new URLClassLoader(jarUrls.toArray(new URL[0]), this.getClass().getClassLoader())
+            );
```

Comment:
Do we really need to do this as a privileged action?. Constructor of `URLClassLoader` calls the security manager, thus it may be necessary. Problem is whole `AccessController` class is now deprecated for removal ([ref](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/security/AccessController.html#doPrivileged(java.security.PrivilegedAction))), along with `SecurityManager`. Therefore at least put a comment saying why it is needed

Reply:
Had this block previously to remove the spot bugs warning. will replace it with the static method `URLClassLoader.newInstance()` to obtain the class loader instead.